src/otx/backend/native/engine.py

```diff
@@ -877,13 +875,18 @@ def _apply_param_overrides(self, param_kwargs: dict[str, Any]) -> None:
         """Apply parameter overrides based on the current local variables."""
         sig = inspect.signature(self.train)
         add_kwargs = param_kwargs.pop("kwargs", {})
-        self._cache.update(**add_kwargs)
         for param_name, param in sig.parameters.items():
-            if param_name in param_kwargs:
-                current_value = param_kwargs[param_name]
-                # Apply override if current value matches default and we have an override
-                if (current_value != param.default) or (param_name not in self._cache.args):
+            if param_name in param_kwargs and param_name in self._cache.args:
+                # if both `param_kwargs` and `_cache.args` have the same parameter,
+                # we will use the value from `param_kwargs` if it is different from the default
+                # value of the parameter.
+                # Otherwise, we will keep the value from `_cache.args`.
+                current_value = param_kwargs.pop(param_name)
+                if current_value != param.default:
```

Comment:
Thanks Kirill. I approved the PR, it's a tricky bug. I think when we removed train.yml: https://github.com/open-edge-platform/training_extensions/pull/4369/files we forgot to add this back 

```
overrides:
  reset: data.train_subset.transforms
  precision: 16
  max_epochs: 100
  gradient_clip_val: 35.0
```